packages/lesswrong/components/common/GlobalHotkeys.tsx

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import { registerComponent } from '../../lib/vulcan-lib';
+import { useGlobalKeydown } from './withGlobalKeydown';
+import { useSetTheme, useConcreteThemeOptions } from '../themes/useTheme';
+import { useUpdateCurrentUser } from '../hooks/useUpdateCurrentUser';
+import { useCurrentUser } from './withUser';
+import { userIsAdmin } from '../../lib/vulcan-users';
+
+export const GlobalHotkeys = () => {
+  const currentThemeOptions = useConcreteThemeOptions();
+  const setTheme = useSetTheme();
+  const currentUser = useCurrentUser();
+  const updateCurrentUser = useUpdateCurrentUser();
+
+  useGlobalKeydown((e) => {
+    // Toggle Dark Mode
+    // option+shift+d (mac) / alt+shift+d (everyone else)
+    if (e.altKey && e.shiftKey && !e.ctrlKey && !e.metaKey && e.keyCode === 68) {
+      e.preventDefault();
+
+      const newThemeName = currentThemeOptions.name === 'dark' ? 'default' : 'dark';
+      setTheme({ ...currentThemeOptions, name: newThemeName });
+    }
```

Comment:
This doesn't persist the change, so I think it would be confusing for users - I guess I would prefer that this part is admin-gated on the EA Forum?

Reply:
I think this persists the change for forums other than the EA forum via cookie, while the EA forum also [sets a user setting](https://github.com/ForumMagnum/ForumMagnum/blob/master/packages/lesswrong/components/themes/ThemePickerMenu.tsx#L44).  Do you want me to add that update here as well (gated the same way as it is in the menu), or just admin-gate this for the EA forum without the user field update?  I'm not actually sure if the EA forum looks at the cookie at all, and what inconsistent state between the user setting and cookie might cause.

Reply:
Oh right, we do it differently. We don't use the cookie. Honestly I'm not sure what the non-admin use case for this feature is, and I don't want users to accidentally unset "auto" (I think that's our default). I'd say just admin-gate this for EAF.

Reply:
(We delete the cookie, see `ThemeContextProvider`, but this keyboard shortcut does change the UI so it could be useful for testing purposes)

Reply:
Ok, I've admin-gated that one for the EA forum